frame/system/src/lib.rs

```diff
@@ -1448,6 +1451,24 @@ impl<T: Config> Pallet<T> {
 		})
 	}
 
+	/// Get the current events deposited by the runtime.
+	///
+	/// NOTE: This should only be used in tests. Reading events from the runtime can have a large
+	/// impact on the PoV size of a block. Users should use alternative and well bounded storage
+	/// items for any behavior like this.
+	#[cfg(any(feature = "std", feature = "runtime-benchmarks", test))]
+	pub fn events() -> Vec<EventRecord<T::Event, T::Hash>> {
+		Self::read_events_i_know_what_i_am_doing()
+	}
+
+	/// Get the current events deposited by the runtime.
+	///
+	/// Should only be called if you know what you are doing and outside of the runtime else it can
+	/// have a large impact on the PoV size of a block.
```

Comment:
it should be precised outside of the runtime block execution or something. Because offchain workers are inside the runtime.